Client/Frontend/Settings/CustomSearchEngineForm.swift

```diff
@@ -0,0 +1,188 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+import Shared
+import SnapKit
+import Storage
+import WebImage
+
+private let log = Logger.browserLogger
+
+class CustomSearchEngineForm: SettingsTableViewController {
+    
+    private var urlString: String?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Add Custom Search Engine"
+    }
+    
+    func faviconFor(url: NSURL) -> Favicon? {
+        var resultFavicon: Favicon?
```

Comment:
You are using deffered partially here. You also need to make faviconFor return a deffered result. So that when the favicon is fetched your code knows when its availible. Look for other places where this favicon fetcher was used for examples